examples/platform/linux/AppMain.h

```diff
@@ -28,7 +29,47 @@
 #include "Options.h"
 
 int ChipLinuxAppInit(int argc, char * const argv[], chip::ArgParser::OptionSet * customOptions = nullptr);
-void ChipLinuxAppMainLoop();
+
+/**
+ * A main loop implementation descripts how an application main loop is to be
+ * run:
+ *    - how to execute the main loop
+ *    - what to do to stop it (inside a signal handler - CTRL+C is captured
+ *      by the main loop function)
+ */
+class AppMainLoopImplementation
+{
+public:
+    virtual ~AppMainLoopImplementation() = default;
+    /**
+     * Execute main loop. Generally should have at least some
+     * `DeviceLayer::PlatformMgr().RunEventLoop();` or equivalent setup
+     */
+    virtual void RunMainLoop() = 0;
```

Comment:
This should probably be documented as not returning until the main loop terminates (and in particular until SignalSafeStopMainLoop is called).